c/storage_mem.c

```diff
@@ -0,0 +1,286 @@
+/*
```

Comment:
Perhaps this file belongs in the `test/` directory?

Reply:
I would leave it here until we use a test suite.